src/modules/user/apiKeys/sagas/apiKeysSaga.ts

```diff
@@ -1,16 +1,21 @@
 // tslint:disable-next-line
 import { call, put } from 'redux-saga/effects';
 import { API, RequestOptions } from '../../../../api';
-import { alertPush } from '../../../public/alert';
+import { alertPush, getCsrfToken } from '../../../index';
 import { apiKeys2FAModal, apiKeysData, ApiKeysFetch } from '../actions';
 
-const apiKeysOptions: RequestOptions = {
-    apiVersion: 'barong',
-};
+const apiKeysOptions = (csrfToken?: string): RequestOptions => (
+    csrfToken ? {
+        apiVersion: 'barong',
+        headers: { 'X-CSRF-Token': csrfToken },
+    } : {
+        apiVersion: 'barong',
+    });
 
 export function* apiKeysSaga(action: ApiKeysFetch) {
     try {
-        const apiKeys = yield call(API.get(apiKeysOptions), '/resource/api_keys');
+        const currentCsrfToken = yield getCsrfToken();
+        const apiKeys = yield call(API.get(apiKeysOptions(currentCsrfToken)), '/resource/api_keys');
```

Comment:
POST, PUT, PATCH, DELETE and TRACE verbs, being state changing verbs, should have a CSRF token attached to the request